### PR TITLE
Add upgrader api endpoints

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: v4/source/system.yaml [hunk 1]
```diff
@@ -1095,6 +1095,89 @@
...
+            application/json:
+              schema:
+                type: object
+                properties:
+                  percentage:
+                    type: integer
+                    description: Current percentage of the upgrade
+                  error:
+                    type: string
+                    description: Error happend during the upgrade
```
Comment (catalintomai) [inline v4/source/system.yaml:1153]: typo

File: v4/source/system.yaml [hunk 2]
```diff
@@ -1095,6 +1095,89 @@
...
+        "403":
+          $ref: "#/components/responses/Forbidden"
+
+  /restart:
+    post:
+      tags:
+        - system
+      summary: Restart the system after an upgrade from Team Edition to Enterprise Edition
+      description: >
+        It restart the current running mattermost instance to execute the new
```
Comment (catalintomai) [inline v4/source/system.yaml:1163]: typo

File: v4/source/system.yaml [hunk 3]
```diff
@@ -1095,6 +1095,89 @@
...
+          $ref: "#/components/responses/ToManyRequests"
+
+  /upgrade_to_enterprise/status:
+    get:
+      tags:
+        - system
+      summary: Get the current status for the implace upgrade from Team Edition to Enterprise Edition
+      description: >
+        It returns the percentage of completion of the current upgrade or the
+        error if, there is any.
```
Comment (catalintomai) [inline v4/source/system.yaml:1133]: remove

File: v4/source/system.yaml [hunk 4]
```diff
@@ -1095,6 +1095,89 @@
...
+        "202":
+          description: Upgrade started
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PushNotification"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "429":
+          $ref: "#/components/responses/ToManyRequests"
```
Comment (catalintomai) [inline v4/source/system.yaml:1124]: Too instead of To

File: v4/source/system.yaml [hunk 5]
```diff
@@ -1095,6 +1095,89 @@
...
+
+  /upgrade_to_enterprise/status:
+    get:
+      tags:
+        - system
+      summary: Get the current status for the implace upgrade from Team Edition to Enterprise Edition
```
Comment (catalintomai) [inline v4/source/system.yaml:1130]: typo

File: v4/source/definitions.yaml [hunk 6]
```diff
@@ -41,6 +41,12 @@ components:
         application/json:
           schema:
             $ref: "#/components/schemas/AppError"
+    ToManyRequests:
+      description: To many requests
```
Comment (catalintomai) [inline v4/source/definitions.yaml:45]: Too

Comment (catalintomai) [inline v4/source/definitions.yaml:45]: Too?

Comment (jespino) [reply v4/source/definitions.yaml]: Done

File: v4/source/system.yaml (hunk 4 above)
Comment (jespino) [reply v4/source/system.yaml]: done

File: v4/source/system.yaml (hunk 5 above)
Comment (jespino) [reply v4/source/system.yaml]: done

File: v4/source/system.yaml (hunk 3 above)
Comment (jespino) [reply v4/source/system.yaml]: done

File: v4/source/system.yaml (hunk 1 above)
Comment (jespino) [reply v4/source/system.yaml]: done

File: v4/source/system.yaml (hunk 2 above)
Comment (jespino) [reply v4/source/system.yaml]: done

**Pull request merged**